packages/valory/contracts/service_registry/contract.py

```diff
@@ -82,15 +84,23 @@ def verify_contract(
         :param contract_address: the contract address
         :return: the verified status
         """
+        verified = False
         ledger_api = cast(EthereumApi, ledger_api)
+        chain_id = ledger_api.api.eth.chain_id
+        expected_address = EXPECTED_CONTRACT_ADDRESS_BY_CHAIN_ID[chain_id]
+        if contract_address != expected_address:
+            _logger.error(
+                f"For chain_id {chain_id} expected {expected_address} and got {contract_address}."
+            )
+            return dict(verified=verified)
         deployed_bytecode = ledger_api.api.eth.get_code(contract_address).hex()
         sha512_hash = hashlib.sha512(deployed_bytecode.encode("utf-8")).hexdigest()
-        verified = DEPLOYED_BYTECODE_MD5_HASH == sha512_hash
+        verified = DEPLOYED_BYTECODE_MD5_HASH_BY_CHAIN_ID[chain_id] == sha512_hash
         if not verified:
-            log_msg = "CONTRACT NOT VERIFIED! reason: frequent changes."
-            log_msg += f". Verified: {verified}. Contract address: {contract_address}."
-            _logger.error(f"{log_msg} Address: {CHAIN_ADDRESS}, chain_id: {CHAIN_ID}")
-        return dict(verified=verified, sha512_hash=sha512_hash)
+            _logger.error(
+                f"CONTRACT NOT VERIFIED! Contract address: {contract_address}, chain_id: {chain_id}."
+            )
```

Comment:
we're no longer return `True` by default as we did when the contract bytecode was continuously changing, we may actually return `False` as well, so this log message is incorrect.

Reply:
I think you missed the `if not verified` condition above.

Reply:
you are right. Lets change the message to be more explicit, I used this to indicate that verification was not performed, in this case it has been performed but failed. e.g. `BYTECODE VERIFICATION FAILED` or something like that

Reply:
Opened #1280 because this is out of the scope of this PR.